aea/cli/generate_all_protocols.py

```diff
@@ -0,0 +1,594 @@
+    return map(methodcaller("absolute"), protocols)
+
+
+def log(message: str, level: int = logging.INFO) -> None:
+    """Produce a logging message."""
+    logger.log(level, message)
+
+
+def log_separator(length: int = 100) -> None:
+    """Log separator."""
+    click.echo("=" * length)
+
+
+def run_cli(*args: Any, **kwargs: Any) -> None:
+    """Run a CLI command."""
+    log(f"Calling command {args} with kwargs {kwargs}")
+    return_code = subprocess.check_call(args, **kwargs)  # nosec
+    enforce(
+        return_code == 0,
+        f"Return code of {pprint.pformat(args)} is {return_code} != 0.",
+    )
+
+
+def run_aea(*args: Any, **kwargs: Any) -> None:
+    """
+    Run an AEA command.
+
+    :param args: the AEA command
+    :param kwargs: keyword arguments to subprocess function
+    """
+    run_cli(sys.executable, "-m", "aea.cli", *args, **kwargs)
+
+
+class AEAProject:
+    """A context manager class to create and delete an AEA project."""
+
+    old_cwd: str
+    temp_dir: str
+
+    def __init__(self, name: str = "my_aea", parent_dir: Optional[str] = None):
+        """
+        Initialize an AEA project.
+
+        :param name: the name of the AEA project.
+        :param parent_dir: the parent directory.
+        """
+        self.name = name
+        self.parent_dir = parent_dir
+
+    def __enter__(self) -> None:
+        """Create and enter into the project."""
+        self.old_cwd = os.getcwd()
+        self.temp_dir = tempfile.mkdtemp(dir=self.parent_dir)
+        os.chdir(self.temp_dir)
+
+        run_aea("create", "--local", "--empty", self.name, "--author", "fetchai")
+        os.chdir(self.name)
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:  # type: ignore
+        """Exit the context manager."""
+        os.chdir(self.old_cwd)
+        shutil.rmtree(self.temp_dir)
+
+
+def _save_specification_in_temporary_file(
+    name: str, specification_content: str
+) -> None:
+    """
+    Save the specification in a temporary file.
+
+    :param name: the name of the package.
+    :param specification_content: the specification content.
+    """
+    # here, the cwd is the temporary AEA project
+    # hence, we are writing in a temporary directory
+    spec_path = Path("..", name + ".yaml")
+    log(f"Save specification '{name}' in temporary file {spec_path}")
+    spec_path.write_text(specification_content)
+
+
+def _generate_protocol(package_path: Path) -> None:
+    """
+    Generate the protocol.
+
+    :param package_path: package to the path.
+    """
+    cmd = ["generate", "protocol", os.path.join("..", package_path.name) + ".yaml"]
+    log(f"Generate the protocol. Command: {pprint.pformat(cmd)}")
+    run_aea(*cmd)
+
+
+def run_isort_and_black(directory: Path, **kwargs: Any) -> None:
+    """Run black and isort against a directory."""
+    kwargs["stdout"] = subprocess.PIPE
+    run_cli(
+        sys.executable, "-m", "black", "-q", str(directory.absolute()), **kwargs,
+    )
+    run_cli(
```

Comment:
Should probably first check if installed, if not ask user to install, for nicer ux